src/igl/opengl/UniformAdapter.h

```diff
@@ -51,7 +51,7 @@ class UniformAdapter {
   uint32_t maxUniforms_ = 1024;
 
   // map for uniform binding indices to the buffers
-  std::unordered_map<int, std::pair<IBuffer*, size_t>> uniformBufferBindingMap_;
+  std::unordered_map<int, std::tuple<IBuffer*, size_t, size_t>> uniformBufferBindingMap_;
```

Comment:
Please declare a local struct right above this line with distinct names for all member fields.

Reply:
add new struct : UniformBufferRange